PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/PrebidNativeAd.java

```diff
@@ -134,101 +142,127 @@ public void onCacheExpired() {
                         }
                     }
                 }
-                return ad.isValid() ? ad : null;
+                return ad;
             } catch (JSONException e) {
                 e.printStackTrace();
             }
         }
         return null;
     }
 
-    private boolean isValid() {
-        return URLUtil.isValidUrl(clickUrl) &&
-                !TextUtils.isEmpty(title) &&
-                !TextUtils.isEmpty(description) &&
-                URLUtil.isValidUrl(iconUrl) &&
-                URLUtil.isValidUrl(imageUrl) &&
-                !TextUtils.isEmpty(cta);
-    }
-
     private PrebidNativeAd() {
     }
 
-    private void setClickUrl(String clickUrl) {
-        this.clickUrl = clickUrl;
+    public void addTitle(NativeTitle title) {
+        titles.add(title);
     }
 
-    private void setTitle(String title) {
-        this.title = title;
+    public void addData(NativeData data) {
+        dataList.add(data);
     }
 
-    private void setDescription(String description) {
-        this.description = description;
+    public void addImage(NativeImage image) {
+        images.add(image);
     }
 
-    private void setIconUrl(String iconUrl) {
-        this.iconUrl = iconUrl;
+    public ArrayList<NativeTitle> getTitles() {
+        return titles;
     }
 
-    private void setImageUrl(String imageUrl) {
-        this.imageUrl = imageUrl;
+    public ArrayList<NativeImage> getImages() {
+        return images;
     }
 
-    private void setCallToAction(String cta) {
-        this.cta = cta;
+    public ArrayList<NativeData> getDataList() {
+        return dataList;
     }
 
-    private void setSponsoredBy(String sponsoredBy) {
-        this.sponsoredBy = sponsoredBy;
+    private void setClickUrl(String clickUrl) {
+        this.clickUrl = clickUrl;
     }
 
     /**
      * @return String title of the Native Ad
-     * */
+     */
+    @Nullable
```

Comment:
We should keep the same API for existing properties. Need to remove `@Nullable` and return empty values. This is relevant to all properties.